server/raft.go

```diff
@@ -1654,7 +1654,7 @@ func (n *raft) State() RaftState {
 func (n *raft) Progress() (index, commit, applied uint64) {
 	n.RLock()
 	defer n.RUnlock()
-	return n.pindex + 1, n.commit, n.applied
```

Comment:
This was an off-by-one, but this value was not used within the code. Now we started to use its value.

`n.pindex == n.commit == n.appplied` means we have a log where every entry is applied.
If we have `n.pindex` 1 greater than `n.commit` that means we have one uncommitted entry. If we have `n.pindex` 1 greater than `n.applied` but equal to `n.commit`, that means we have a committed entry that's not applied yet.